external/wakaama/examples/client/lwm2mclient.c

```diff
@@ -1209,48 +1210,55 @@ int lwm2m_client_main(int argc, char *argv[])
     if (NULL == objArray[1])
     {
         fprintf(stderr, "Failed to create server object\r\n");
+		close(data.sock);
         return -1;
     }
 
     objArray[2] = get_object_device();
     if (NULL == objArray[2])
     {
         fprintf(stderr, "Failed to create Device object\r\n");
+		close(data.sock);
         return -1;
     }
 
     objArray[3] = get_object_firmware();
     if (NULL == objArray[3])
     {
         fprintf(stderr, "Failed to create Firmware object\r\n");
+		close(data.sock);
         return -1;
     }
 
     objArray[4] = get_object_location();
     if (NULL == objArray[4])
     {
         fprintf(stderr, "Failed to create location object\r\n");
+		close(data.sock);
         return -1;
     }
 
     objArray[5] = get_test_object();
     if (NULL == objArray[5])
     {
         fprintf(stderr, "Failed to create test object\r\n");
+		close(data.sock);
         return -1;
     }
 
     objArray[6] = get_object_conn_m();
     if (NULL == objArray[6])
     {
         fprintf(stderr, "Failed to create connectivity monitoring object\r\n");
+		close(data.sock);
         return -1;
```

Comment:
Please, check final source codes for external/wakaama/examples/client/lwm2mclient.c. Do not have these codes. We have changed back wakama to it's original state. I mean original opensource.